### PR TITLE
Autocorrect UI improvements

### PR DESCRIPTION
**Pull request by AbigailBuccaneer**
* Make autocorrect prettier:
image

* Add an Atom/IntelliJ autocomplete style. (I'm losing my mind without this!)

First commit is luacheck issue fixes so you might wanna skip it when reviewing.

### CODE REVIEW

File: lua/wire/client/text_editor/texteditor.lua [hunk 1]
```diff
@@ -1154,14 +1159,16 @@ function EDITOR:CountFinds( str )
 	if whole_word_only then
 		local pattern = "([^a-zA-Z0-9_])"..str.."([^a-zA-Z0-9_])"
 		txt = " " .. txt
-		local num1, num2 = 0, 0
+		local num1, num2
 		txt, num1 = txt:gsub( pattern, "%1%2" )
 		if txt == "" then return num1 end
-		txt, num2 = txt:gsub( pattern, "%1%2" )
+		local _
+		_, num2 = txt:gsub( pattern, "%1%2" )
```
Comment (bigdogmat) [inline lua/wire/client/text_editor/texteditor.lua]: `num2 = select(2, txt:gsub(pattern, "%1%2"))` might look a bit better here

File: lua/wire/client/text_editor/texteditor.lua [hunk 2]
```diff
@@ -2634,7 +2635,7 @@ function EDITOR:AC_FillList()
 	panel.Selected = 0
 	local maxw = 15
 
-	surface.SetFont( "E2SmallFont" )
+	surface.SetFont( self.CurrentFont )
```
Comment (bigdogmat) [inline lua/wire/client/text_editor/texteditor.lua]: Spaces

File: lua/wire/client/text_editor/texteditor.lua (hunk 1 above)
Comment (Divran) [reply lua/wire/client/text_editor/texteditor.lua]: or `num2 = {whatever}[2]` idk which is faster

Comment (bigdogmat) [reply lua/wire/client/text_editor/texteditor.lua]: That has the overhead of table creation. Also as a bonus `select` is compiled when first argument is a constant.

File: lua/wire/client/text_editor/texteditor.lua [hunk 3]
```diff
@@ -2647,13 +2648,16 @@ function EDITOR:AC_FillList()
...
+			else
+				backgroundColor = Color(65, 105, 225, 255)
 			end
-			-- I honestly dont have a fucking clue.
-			-- h2, was being cleaned up instantly for no reason.
-			surface.SetFont( "E2SmallFont" )
+			surface_SetDrawColor(backgroundColor)
+			surface_DrawRect(0, 0, w, h)
+
+			surface.SetFont( self.CurrentFont )
```
Comment (bigdogmat) [inline lua/wire/client/text_editor/texteditor.lua]: Spaces

**Comment by AbigailBuccaneer** [comment]
@bigdogmat review comments addressed

**Pull request merged**